air_flags/rollout.py

```diff
@@ -0,0 +1,128 @@
+import random
+from dataclasses import dataclass
+from datetime import datetime
+from typing import Optional
+
+ROLLOUT_VALUES = [True, False]
+DATE_FORMAT = "%Y-%m-%d"
+CANARY_STRATEGY = "canary"
+SCHEDULED_STRATEGY = "scheduled"
+PROGRESSIVE_STRATEGY = "progressive"
+ROLLOUT_STRATEGIES = [
+    CANARY_STRATEGY,
+    SCHEDULED_STRATEGY,
+    PROGRESSIVE_STRATEGY,
+]
+MIN_PERCENTAGE = 1
+MAX_PERCENTAGE = 100
+
+
+@dataclass
+class Rollout:
+    """Rollout entity"""
+
+    strategy: str
+    percentage: Optional[int] = None
+    start_date: Optional[str] = None
+    end_date: Optional[str] = None
+
+    def __post_init__(self):
+        if self.strategy not in ROLLOUT_STRATEGIES:
+            raise ValueError(f"Field 'strategy' must be {ROLLOUT_STRATEGIES}")
+        if self.percentage is not None and self.percentage not in range(
+            MIN_PERCENTAGE, MAX_PERCENTAGE + 1
+        ):
+            raise ValueError("Field 'percentage' must be between 1 and 100")
+        if self.start_date:
+            try:
+                datetime.strptime(str(self.start_date), DATE_FORMAT)
+            except ValueError:
+                raise ValueError(
+                    "Field 'start_date' must be a date with the"
+                    " format 'YYYY-mm-dd'"
+                )
+        if self.end_date:
+            try:
+                datetime.strptime(str(self.end_date), DATE_FORMAT)
+            except ValueError:
+                raise ValueError(
+                    "Field 'end_date' must be a date with the"
+                    " format 'YYYY-mm-dd'"
+                )
+
+    @property
+    def value(self) -> bool:
+        if not self._is_on_date():
+            return False
+        if self.strategy == CANARY_STRATEGY:
+            return self._gen_canary_value()
+        elif self.strategy == SCHEDULED_STRATEGY:
+            return self._gen_scheduled_value()
+        elif self.strategy == PROGRESSIVE_STRATEGY:
+            return self._gen_progressive_value()
```

Comment:
You can remove `else` statement
```suggestion
        if not self._is_on_date():
            return False
        if self.strategy == CANARY_STRATEGY:
            return self._gen_canary_value()
        if self.strategy == SCHEDULED_STRATEGY:
            return self._gen_scheduled_value()
        if self.strategy == PROGRESSIVE_STRATEGY:
            return self._gen_progressive_value()
```